contracts/governance/Governor.sol

```diff
@@ -538,19 +548,22 @@ abstract contract Governor is Context, ERC165, EIP712, IGovernor, IERC721Receive
     function castVoteWithReasonAndParamsBySig(
         uint256 proposalId,
         uint8 support,
+        address voter,
         string calldata reason,
         bytes memory params,
         uint8 v,
         bytes32 r,
         bytes32 s
```

Comment:
Given that we are changing the signature of the `castVoteBySig` functions, what do you think about changing these arguments to `bytes signature` and using the opportunity to add EIP-1271 compatibility?

I don't think we should do this here so it's not a blocker for this PR.

Reply:
This also aligns really well with the addition of the `voter` argument, which is needed for EIP-1271.

Reply:
According to what we discussed, I'll add the signature changes and 1271 compatibility in a follow up PR